### PR TITLE
Initial stub out of site layout.

### PR DESCRIPTION
**Pull request by ScottDowne**
@mmmavis the WIP so far.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,9 +51,16 @@
     "webpack": "^1.14.0"
   },
   "dependencies": {
+    "html-loader": "^0.4.4",
+    "json-loader": "^0.5.4",
+    "loader-utils": "^0.2.16",
+    "markdown-loader": "^0.1.7",
+    "markdown-react-js": "^0.3.0",
+    "marked": "^0.3.6",
```
Comment (ScottDowne) [inline package.json]: I think I can move these to dev dependencies? Havn't taken a close look yet.

File: source/js/components/details.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+var React = require(`react`);
```
Comment (mmmavis) [inline source/js/components/details.js]: `import React from 'react';` ?

File: source/js/components/details.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        </Panel>
+        <Panel activeKey={this.state.activeKey} activateKey={this.onKeyChange} itemKey="item-6" header="Challenge rules">
+          <ReactMarkdown source={item6}/>
+        </Panel>
+        <Panel activeKey={this.state.activeKey} activateKey={this.onKeyChange} itemKey="item-7" header="FAQ">
+          <ReactMarkdown source={item7}/>
+        </Panel>
+        <Panel activeKey={this.state.activeKey} activateKey={this.onKeyChange} itemKey="item-8" header="Getting Started">
+          <ReactMarkdown source={item8}/>
+        </Panel>
```
Comment (mmmavis) [inline source/js/components/details.js]: There are a bunch of repetitions here. I wonder if we should create an array of objects. Each object just stores values of the `itemKey`, `header`, and `source`. We can then loop through the array using `.map()` to generate the list of `<Panel>`s.

File: source/js/components/sticky-container.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  componentDidMount: function() {
+    document.addEventListener(`scroll`, this.onScroll);
+    setTimeout(this.onScroll);
+  },
+  componentWillUnmount: function() {
+    document.removeEventListener(`scroll`, this.onScroll);
+  },
+  render: function() {
+    var className = this.props.className || "";
```
Comment (mmmavis) [inline source/js/components/sticky-container.js]: Let's rely on the [`classnames`](https://github.com/JedWatson/classnames) npm package to handle classnames related stuff. We use this package for many other MoFo sites... it's quite neat!

File: source/pug/partials/nav.pug [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#sticky-nav
+
+script(src=`${env.virtualRoot}_js/main.compiled.js`)
```
Comment (mmmavis) [inline source/pug/partials/nav.pug]: Wait do we need to load `main.compiled.js` here as well? 🙄 Loading it in `source/pug/views/home.pug` should be sufficient I suppose.

**Comment by mmmavis** [comment]
@ScottDowne I skimmed through the code just to understand how things work. Things look great 😃 👍 . It needs tweaks here and there but they're mostly minor improvements.

2 slightly bigger things: 
I think we should start using the `classnames` npm package (see [my comment](https://github.com/mozilla/nsf-challenge/pull/1/files#r98574119)) and reduce code repetitions (see [comment](https://github.com/mozilla/nsf-challenge/pull/1/files#r98573191)).

Markdown wise I'd like to defer it to @gvn to review as I don't have any experience setting it up.

**Comment by gvn** [comment]
Markdown is fine and makes sense for this site given that it's mostly large blocks of non-localized text. 👍

File: source/js/components/details.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    if (key !== this.state.activeKey) {
+      this.setState({
+        activeKey: key
+      });
+    }
+  },
+  render: function() {
+    return (
+      <Expander>
+        <Panel activeKey={this.state.activeKey} activateKey={this.onKeyChange} itemKey="item-1" header="How to enter the challenge">
```
Comment (ScottDowne) [inline source/js/components/details.js]: @mmmavis Is it this bit you want generated dynamically?

**Review by mmmavis — CHANGES_REQUESTED** [review]

File: source/pug/partials/nav.pug (hunk 5 above)
Comment (ScottDowne) [reply source/pug/partials/nav.pug]: We do not need this, I just missed it.

**Comment by ScottDowne** [comment]
@mmmavis I believe that's all of them fixed.

**Comment by mmmavis** [comment]
Thanks @ScottDowne ! Let's fix the Travis error: https://travis-ci.org/mozilla/nsf-challenge/builds/197071170 and have this PR merged so it doesn't grow tooooooo bigggggggggg. 😀

**Comment by ScottDowne** [comment]
k, done, should be building soon.

**Comment by ScottDowne** [comment]
@mmmavis finally got this passing.

**Review by mmmavis — APPROVED** [review]
@ScottDowne I think we can merge this one first and tackle rest of the work in followup PRs.

**Comment by ScottDowne** [comment]
Cool. Feel free to hit merge as I'm afk for a few.

**Pull request merged**